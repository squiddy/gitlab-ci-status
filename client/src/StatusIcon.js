import React from "react";

import failed from "./images/failed.svg";
import success from "./images/success.svg";
import skipped from "./images/skipped.svg";
import running from "./images/running.svg";
import created from "./images/created.svg";
import pending from "./images/pending.svg";

export function StatusIcon({ status, title }) {
  // Renders a visual icon for the build/pipeline status.
  const url = {
    failed,
    success,
    skipped,
    canceled: skipped,
    running,
    created,
    pending
  }[status];
  return <img title={title} alt="" src={url} />;
}
