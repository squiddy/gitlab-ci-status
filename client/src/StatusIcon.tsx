import React, { ClassAttributes, SVGAttributes } from 'react';

import { ReactComponent as Failed } from './images/failed.svg';
import { ReactComponent as Success } from './images/success.svg';
import { ReactComponent as Skipped } from './images/skipped.svg';
import { ReactComponent as Running } from './images/running.svg';
import { ReactComponent as Created } from './images/created.svg';
import { ReactComponent as Pending } from './images/pending.svg';
import { Status } from './types';

const Components: { [index: string]: React.FunctionComponent } = {
  [Status.Failed]: Failed,
  [Status.Success]: Success,
  [Status.Skipped]: Skipped,
  [Status.Canceled]: Skipped,
  [Status.Running]: Running,
  [Status.Created]: Created,
  [Status.Pending]: Pending,
  [Status.Manual]: Skipped
};

// FIXME Using any here on the props, because I have no idea how to define all
// possible SVG attributes as props.
export function StatusIcon({ status, ...rest }: { status: Status } & any) {
  // Renders a visual icon for the build/pipeline status.
  const Component = Components[status];
  return <Component {...rest} />;
}
