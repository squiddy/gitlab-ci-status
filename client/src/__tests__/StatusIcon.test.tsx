import React from "react";
import renderer from "react-test-renderer";

import { StatusIcon } from "../StatusIcon";

it("StatusIcon renders correctly", () => {
  const tree = renderer
    .create(<StatusIcon status="failed" title="Failed" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
