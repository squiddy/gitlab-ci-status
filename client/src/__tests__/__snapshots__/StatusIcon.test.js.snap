// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon renders correctly 1`] = `
<img
  alt=""
  src="failed.svg"
  title="Failed"
/>
`;
